synapse/rest/admin/federation.py

```diff
@@ -111,12 +111,26 @@ async def on_GET(
     ) -> Tuple[int, JsonDict]:
         await assert_requester_is_admin(self._auth, request)
 
+        if not await self._store.is_destination_known(destination):
+            raise NotFoundError("Unknown destination")
+
         destination_retry_timings = await self._store.get_destination_retry_timings(
             destination
         )
 
-        if not destination_retry_timings:
-            raise NotFoundError("Unknown destination")
+        retry_timing_response: JsonDict = {}
+        if destination_retry_timings:
+            retry_timing_response = {
+                "failure_ts": destination_retry_timings.failure_ts,
+                "retry_last_ts": destination_retry_timings.retry_last_ts,
+                "retry_interval": destination_retry_timings.retry_interval,
+            }
+        else:
+            retry_timing_response = {
+                "failure_ts": None,
+                "retry_last_ts": 0,
+                "retry_interval": 0,
+            }
 
         last_successful_stream_ordering = (
```

Comment:
This query only makes sense if the destination is known, I think?

Reply:
Yes, of course. If destination is unknown it is aborted in line 114/115:
```python
        if not await self._store.is_destination_known(destination):
            raise NotFoundError("Unknown destination")
```

Reply:
Oh right! Not sure what I was thinking...